openpype/hosts/maya/api/plugin.py

```diff
@@ -143,16 +143,49 @@ def load(
         assert os.path.exists(self.fname), "%s does not exist." % self.fname
 
         asset = context['asset']
+        subset = context['subset']
+        settings = get_project_settings(context['project']['name'])
+        custom_naming = settings['maya']['load']['reference_loader']
         loaded_containers = []
 
-        count = options.get("count") or 1
-        for c in range(0, count):
-            namespace = namespace or lib.unique_namespace(
-                "{}_{}_".format(asset["name"], context["subset"]["name"]),
-                prefix="_" if asset["name"][0].isdigit() else "",
-                suffix="_",
+        if not custom_naming['namespace']:
+            raise ValueError("No namespace specified")
+        elif not custom_naming['group_name']:
+            raise ValueError("No group name specified")
+
+        custom_namespace = custom_naming['namespace'].format(
+            asset_name=asset['name'],
+            asset_type=asset['type'],
+            subset=subset['name'],
+            family=(
+                subset['data'].get('family') or
+                subset['data']['families'][0]
+            )
+        )
+
+        custom_group_name = custom_naming['group_name'].format(
+            asset_name=asset['name'],
+            asset_type=asset['type'],
+            subset=subset['name'],
+            family=(
+                subset['data'].get('family') or
+                subset['data']['families'][0]
             )
+        )
 
+        namespace = lib.unique_namespace(
+            custom_namespace,
+            prefix="_" if custom_namespace[0].isdigit() else "",
+            suffix=""
+        )
```

Comment:
This logic **must** remain in the for loop below otherwise it's not ensured that each loaded entry gets a unique namespace.